validator/client/src/main/java/tech/pegasys/teku/validator/client/signer/ExternalSignerBlockRequestProvider.java

```diff
@@ -43,12 +47,24 @@ public Map<String, Object> getBlockMetadata(final Map<String, Object> additional
 
     final tech.pegasys.teku.api.schema.BeaconBlock beaconBlock =
         schemaObjectProvider.getBeaconBlock(block);
-    if (signType == SignType.BLOCK) {
-      metadata.put("block", beaconBlock); // backward compatible with phase0
-    } else {
-      metadata.put(
-          "beacon_block",
-          new BlockRequestBody(spec.atSlot(block.getSlot()).getMilestone(), beaconBlock));
+    final tech.pegasys.teku.api.schema.BeaconBlockHeader beaconBlockHeader =
+        new tech.pegasys.teku.api.schema.BeaconBlockHeader(blockHeader);
+
+    switch (spec.atSlot(block.getSlot()).getMilestone()) {
+      case PHASE0:
+        metadata.put("block", beaconBlock); // backward compatible with phase0
+        break;
+      case ALTAIR:
+        metadata.put(
+            "beacon_block",
+            new BlockRequestBody(spec.atSlot(block.getSlot()).getMilestone(), beaconBlock, null));
```

Comment:
It probably makes sense to have two separate constructors for `BlockRequestBody`, one that takes just a beacon block and one that takes just a header.  May still need the version that takes both for JSON deserialisation but would be nice to clean this up to not have to specify null.

Reply:
Done, introduced overloaded constructor. Also updated unit tests to assert correct fields are populated.